### DISCUSSION TITLE
Unable to recover signer from hash and signature using viem but works for ethers.js

### DISCUSSION THREAD

**Discussion by tomarsachin2271** [wevm/viem#1685] [Question] [answered]
I'm trying to recover the address which signed a hash by using viem's recoverAddress method, but the value it is giving is not the correct value which i'm expecting. But when i use ethers library, it is giving me correct results.

Viem Version is "^1.20.3"

### Example Values
**Hash:** 0x2a87193349a393ba9a914af959b532120dfb2a2bc7aeabd7de344dd711bb92fc
**Signature:** 0x5ae4e70321e29afe793c5d00711c72716febced4e7944233c7788dacb242325e4c6bc072a509d52aebaee8baab7294017ee3cdb94b37da545b29cb678cd0b5a91b

**Expected Signer Address:** 0xC6dAB8652E5E9749523bA948F42d5944584E4e73

**Code with ethers**
```
recoverSignerFromEthers(hash: string, signature: string): string {
    const ethSignedMessageHash = ethers.utils.arrayify(hash);
    return ethers.utils.verifyMessage(ethSignedMessageHash, signature);
}
```

**Code with viem**
```
async recoverSignerFromViem(hash: Hex, signature: Hex): Promise<string> {
    return recoverAddress({
        hash: hash,
        signature: signature
      })
}
```

### Usage

```
const recoveredSignerFromViem = await this.recoverSignerFromViem(hash, signature);
console.debug("Recovered address from Viem: ", recoveredSignerFromViem);

let recoveredSignerFromEthers = this.recoverSignerFromEthers(hash, signature);
console.log("Recovered address from ethers: ", recoveredSignerFromEthers);
```

### Console Output

Recovered address from Viem:  0xefCb4A92Df00e54987CDC9D9e1B98E07A466D608
Recovered address from ethers:  0xC6dAB8652E5E9749523bA948F42d5944584E4e73

**Comment by jxom** [comment, marked as answer]
Looks like the hash uses a EIP-191 prefix. So you would need to use `recoverMessageAddress` (`recoverAddress` is the raw version of this, and `recoverMessageAddress` uses it internally but also adds on the prefix).

```ts
await recoverMessageAddress({
  message: { raw: '0x2a87193349a393ba9a914af959b532120dfb2a2bc7aeabd7de344dd711bb92fc' },
  signature: '0x5ae4e70321e29afe793c5d00711c72716febced4e7944233c7788dacb242325e4c6bc072a509d52aebaee8baab7294017ee3cdb94b37da545b29cb678cd0b5a91b',
})
```

**Reply by tomarsachin2271** [reply]
Thanks. It worked 👍

I would suggest to add this use case as well in the [ethers migration](https://viem.sh/docs/ethers-migration) page to have better visibility.

**Reply by tmm** [reply]
PR welcome!
